crates/type-system/src/ontology/uri/mod.rs

```diff
@@ -153,9 +153,8 @@ impl FromStr for VersionedUri {
 
         Ok(Self::new(
             BaseUri::new(base_uri.to_owned()).map_err(ParseVersionedUriError::InvalidBaseUri)?,
-            version
-                .parse()
-                .map_err(|_| ParseVersionedUriError::InvalidVersion)?,
+            u32::from_str(version)
+                .map_err(|error| ParseVersionedUriError::InvalidVersion(error.to_string()))?,
```

Comment:
How come you're going with a string here? Does `parse` not return a useful type that we can store inside?

Reply:
Ah you replied

Reply:
No worries 🙂